js/piwik.js

```diff
@@ -4012,6 +4060,7 @@ if (typeof Piwik !== 'object') {
              */
             detectBrowserFeatures();
             updateDomainHash();
+            setVisitorIdCookie();
```

Comment:
`setVisitorIdCookie()` is only executed if `Piwik.getTracker(url, siteId)` is actually called this way and provided with a siteId. In all other cases (meaning when the default async tracker is used via `_paq`) this method will not have an effect since the method `setVisitorIdCookie()` will only do something if `configTrackerSiteId` is set. 

The tests succeed when updating the test to eg from
```
var delayedTracker = Piwik.getTracker();
```

to 

```
var delayedTracker = Piwik.getTracker();
delayedTracker.setSiteId(1);
```

which would be more realistic. I think it is rather an accident that this test works because the tests are not atomic and do not reset everything in each test suite. In reality one would have to set at least a `siteId` making everything work again. This would also give developers a chance to do the following if they want:


```
var delayedTracker = Piwik.getTracker();
delayedTracker.disableCookies();
delayedTracker.setSiteId(1);
```